packages/utilities/fast-react-wrapper/src/index.ts

```diff
@@ -321,16 +323,24 @@ export function reactWrapper(
                     { ...props, __forwardedRef: ref } as InternalProps,
                     props?.children
                 )
-        ) as ReactWrapper<TElement, TEvents>;
+        ) as React.ComponentType<any>;
```

Comment:
Curious of @nicholasrice and @EisenbergEffect's thoughts here, but I'd like to avoid the casting here and I wonder if this can be cleaned up to maintain the overall structure...

This would create a new constant for the forward ref, set the display name, and then assign it to the component as `forwardRef` while maintaining the current typings, etc. @procload is the reason we need to assign displayName to both the underlying component and the react.reactComponent instance that forwardRef is anonymous by default?

If we assign to ForwardRef, do we need to assign to the underlying component?

```ts
 const displayName = config.name ?? "UnnamedComponent";

        ReactComponent.displayName = displayName;

        const forwardRef = React.forwardRef(
            (
                props?: ReactWrapperProps<TElement, TEvents>,
                ref?: ReactModule.Ref<unknown>
            ) =>
                React.createElement(
                    ReactComponent,
                    { ...props, __forwardedRef: ref } as InternalProps,
                    props?.children
                )
        );

        forwardRef.displayName = displayName;

        const reactComponent = forwardRef as ReactWrapper<TElement, TEvents>;

        if (!wrappersCache.has(type)) {
            wrappersCache.set(type, new Map<string, any>());
        }
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        wrappersCache
            .get(type)!
            .set(config.name ?? DEFAULT_CACHE_NAME, reactComponent);

        return reactComponent;
    }
```

Reply:
Was chatting with @nicholasrice in another thread and he suggested changing the ReactWrapper type to an interface and adding displayName to that. I pushed up the changes.

Reply:
And to answer your question, if I didn't assign it to the forwardRef it remained anonymous. @miroslavstastny suggested changing the forwardRef in his suggested fixes.

Reply:
If what is returned here is the forwardRef (and not the underlying component) do we need both? Just curious...

Reply:
Not sure if we need both, but this is what I'm seeing if we don't. If we only set the displayName on the ReactComponent, we get:
![image](https://github.com/microsoft/fast/assets/49389/d201e2b5-0ed2-4e0c-a45e-d0225f041201)
![image](https://github.com/microsoft/fast/assets/49389/c7adb5b7-2c36-41c9-85c3-114427ef9b09)

Whereas if we set the displayName on the exported forwardRef reactComponent, we get:
![image](https://github.com/microsoft/fast/assets/49389/9c08fd16-457d-46ab-8de8-3a6b4833731a)
![image](https://github.com/microsoft/fast/assets/49389/9cf5e8bf-07d9-4c08-b595-022dcf46d33c)

I believe a lot of this is in service of developer experience, and not being a React developer myself, I'm fine leaving the solution up to what's best for React developers.

Reply:
@procload sorry, to be clear - I totally get setting it on the forwardRef, I'm wondering if we need it on the internal component...does that show in the tree if we're returning the forwardRef instance?

Reply:
I guess we don't _need_ to, but when we take that off we're left with `ReactComponent` in its place. Again, I'm fine either way.

![image](https://github.com/microsoft/fast/assets/49389/29991bfa-8a66-41bb-a178-2a0f27a2ca92)

Reply:
Updated so we're only setting displayName on the forwardRef!